servicemanager/servicemanager.go

```diff
@@ -342,3 +554,120 @@ func (sm *ServiceManager) prepareServiceFS(imagePath string, gid int) (rootFSDig
 
 	return rootFSDigest, nil
 }
+
+func (sm *ServiceManager) removeDemageServicesFolder(services []ServiceInfo) error {
+	files, err := ioutil.ReadDir(sm.servicesDir)
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+filesLoop:
+	for _, file := range files {
+		fullPath := path.Join(sm.servicesDir, file.Name())
+
+		for _, service := range services {
+			if fullPath == service.ImagePath {
+				continue filesLoop
+			}
+		}
+
+		if err = os.RemoveAll(fullPath); err != nil {
+			log.Printf("Can't remove service directory: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func (sm *ServiceManager) tryAllocateSpace(requiredSize int64) (err error) {
+	sm.Lock()
+
+	if sm.numTryAllocateMemory == 0 {
+		if sm.availableSize, err = GetAvailableSize(sm.servicesDir); err != nil {
+			sm.Unlock()
+
+			return aoserrors.Wrap(err)
+		}
+	}
+
+	for _, service := range sm.pendingRemove {
+		log.Println("Try allocate")
+		sm.Unlock()
+
+		if _, err = sm.removeService(service); err != nil {
+			return aoserrors.Wrap(err)
+		}
+
+		sm.Lock()
+	}
+
+	sm.pendingRemove = nil
+
+	if sm.availableSize < requiredSize {
+		sm.Unlock()
+
+		return aoserrors.Errorf("not enough memory to delete service")
+	}
+
+	sm.availableSize -= requiredSize
```

Comment:
should it be += ? Use `deallocateSpace` function.

Reply:
I think it should be `-=`, since we are allocating memory on partitions